### PR TITLE
[Merged by Bors] - Feat: Port/Data.Seq.Seq

### PR DESCRIPTION
**Pull request by arienmalec**
Port of `Data.Seq.Seq` following Mathlib3 refactor to avoid `Seq` overload.

Down to two issues:

- [x] `bind_ret` goes like Mathlib3 right up to the last line but doesn't close (I've tried the full Mathlib3 `simp` invocation to no avail.
- [x] `bind_assoc` has issues with single `match` lines gumming up the works, which should `dsimp` away but don't

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by arienmalec** [comment]
Down to two issues:

- [ ] `bind_ret` goes like Mathlib3 right up to the last line but doesn't close (I've tried the full Mathlib3 `simp` invocation to no avail.
- [ ] `bind_assoc` has issues with single `match` lines gumming up the works, which should `dsimp` away but don't

File: Mathlib/Data/Seq/Seq.lean [hunk 1]
```diff
@@ -262,16 +269,18 @@ theorem tail_nil : tail (nil : Seq α) = nil :=
...
-theorem nth_tail (s : Seq α) (n) : nth (tail s) n = nth s (n + 1) :=
+theorem get?_tail (s : Seq α) (n) : get? (tail s) n = get? s (n + 1) :=
   rfl
-#align stream.seq.nth_tail Stream'.Seq.nth_tail
+#align stream.seq.nth_tail Stream'.Seq.get?_tail
 
-/-- Recursion principle for sequences, compare with `list.rec_on`. -/
-def recOn {C : Seq α → Sort v} (s : Seq α) (h1 : C nil) (h2 : ∀ x s, C (cons x s)) : C s := by
+/-- Recursion principle for sequences, compare with `List.recOn`. -/
+-- porting note: TODO: use match rather than `induction'`
```
Comment (eric-wieser) [inline Mathlib/Data/Seq/Seq.lean:281]: You should be able to use the `rec_eq_recC` trick we use elsewhere to make this computable.

Comment (arienmalec) [reply Mathlib/Data/Seq/Seq.lean:281]: I'm confused how this will work here since we aren't recursing on `Seq` itself, but on `destruct s`...

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**